### PR TITLE
feat: Sup 9265 perform a supervault test with a 5115 in between 4626s

### PR DESCRIPTION
**Pull request by 0xTimepunk**

### CODE REVIEW

File: src/SuperVault.sol [hunk 1]
```diff
@@ -578,44 +598,87 @@ contract SuperVault is BaseStrategy, ISuperVault {
...
+            data.extraFormData = _prepareDepositExtraFormData(superformIds_, length);
+        }
         req.superformData = data;
     }
 
+    /// @notice Prepares extra form data to allow 5115 underlying superforms to succeed
+    /// @param superformIds_ Array of superform IDs
+    /// @param numberOfSuperforms_ Number of superforms
+    /// @return bytes Encoded extra form data
+    function _prepareDepositExtraFormData(
```
Comment (TamaraRingas) [inline src/SuperVault.sol]: Could perhaps be renamed `_prepareDepositExtra5115FormData()`

**Review by TamaraRingas — APPROVED** [review]
LGTM! 

Someone could alter the the balance by sending `SuperPositions` but I think it may be best to merge this and address that in a subsequent PR

File: src/SuperVault.sol [hunk 2]
```diff
@@ -430,54 +441,127 @@ contract SuperVault is BaseStrategy, ISuperVault {
...
+        vars.spBalances = new uint256[](_numberOfSuperforms_);
+        vars.assetBalances = new uint256[](_numberOfSuperforms_);
+
+        // 1. Snapshot assets and SP balances. This is relevant for withdraws and must be calculated in advance
+        for (uint256 i; i < _numberOfSuperforms_; ++i) {
+            (superform,,) = _superformIds_[i].getSuperform();
+            superformContract = IBaseForm(superform);
+
+            vars.spBalances[i] =
+                ISuperPositions(_getAddress(keccak256("SUPER_POSITIONS"))).balanceOf(address(this), _superformIds_[i]);
```
Comment (gpersoon) [inline src/SuperVault.sol:490]: `_getAddress(keccak256("SUPER_POSITIONS")))` can be cached to improve readbility and safe gas

File: src/SuperVault.sol [hunk 3]
```diff
@@ -430,54 +441,127 @@ contract SuperVault is BaseStrategy, ISuperVault {
...
+                    // This is slippage protected anyway at the form level
+                    mvData.outputAmounts[i] =
+                        isERC5115 ? vars.assetBalances[i] - TOLERANCE_CONSTANT : vars.assetBalances[i];
+                } else {
+                    // For partial withdrawals, use proportional amounts
+                    uint256 amountOut = amount_.mulDiv(weights[i], TOTAL_WEIGHT, Math.Rounding.Down);
+
+                    mvData.outputAmounts[i] = isERC5115 ? amountOut - TOLERANCE_CONSTANT : amountOut;
+
+                    mvData.amounts[i] = superformContract.previewDepositTo(amountOut);
```
Comment (gpersoon) [inline src/SuperVault.sol:531]: shouldn't this be the withdraw version?  (`previewWithdrawFrom()`) ?

File: src/SuperVault.sol [hunk 4]
```diff
@@ -430,54 +441,127 @@ contract SuperVault is BaseStrategy, ISuperVault {
...
+
+                    // If vault is ERC5115, subtract tolerance constant because of minAmountOut check to avoid errors.
+                    // This is slippage protected anyway at the form level
+                    mvData.outputAmounts[i] =
+                        isERC5115 ? vars.assetBalances[i] - TOLERANCE_CONSTANT : vars.assetBalances[i];
+                } else {
+                    // For partial withdrawals, use proportional amounts
+                    uint256 amountOut = amount_.mulDiv(weights[i], TOTAL_WEIGHT, Math.Rounding.Down);
+
+                    mvData.outputAmounts[i] = isERC5115 ? amountOut - TOLERANCE_CONSTANT : amountOut;
```
Comment (gpersoon) [inline src/SuperVault.sol:529]: perhaps have a function to adjust for the tolerance, for easier maintenance, something like:
```solidity
function tolerance(bool isERC5115, uint256 amount) view returns (uint256){
   return isERC5115 ? amount - TOLERANCE_CONSTANT : amount;
}
```

File: src/SuperVault.sol (hunk 2 above)
Comment (0xTimepunk) [reply src/SuperVault.sol:490]: [40ee60d](https://github.com/superform-xyz/SuperVaults/pull/36/commits/40ee60d2be8c582e796a278e1f1b7d64a3221194)

File: src/SuperVault.sol (hunk 3 above)
Comment (0xTimepunk) [reply src/SuperVault.sol:531]: Yearn tokenized strategies provide only "assets" and not shares when redeeming. So, first we have outputAmounts (which in Superform is the output "assets") . Then for it to interact with Superform core we must work back to shares, hence the reason for previewDepositTo

File: src/SuperVault.sol (hunk 4 above)
Comment (0xTimepunk) [reply src/SuperVault.sol:529]: [30f5f82](https://github.com/superform-xyz/SuperVaults/pull/36/commits/30f5f822cc7278e2a841739d1c025252f0fdbd83) added

**Pull request merged**